### ISSUE TITLE
[BUG]: train loss is nan at stage 1

### ISSUE THREAD

**Issue by boundles** [hpcaitech/ColossalAI#3444]
### 🐛 Describe the bug

torchrun --standalone --nproc_per_node=2 train_sft.py \
                --pretrain /nfs/dataset-dtai-common/chatgpt/models/llama-7b-hf \
                --model 'llama' \
                --strategy ddp \
                --log_interval 10 \
                --save_path  /nfs/dataset-dtai-common/chatgpt/models/Coati-7B \
                --dataset /nfs/dataset-dtai-common/chatgpt/data/InstructionWild/instinwild_en.json \
                --batch_size 4 \
                --accumulation_steps 8 \
                --lr 2e-5 \
                --lora_rank 4 \
                --max_datasets_size 512 \
                --max_epochs 1 \

dataset is instinwild_en.json, ddp strategy and lora rank is 4

**Comment by alibabadoufu** [comment]
Facing the same issue

**Comment by JThh** [comment]
Solving the same issue at #3439
